src/global/ValueIdComparators.h

```diff
@@ -462,16 +462,15 @@ ComparisonResult compareIdsImpl(ValueId a, ValueId b, auto comparator) {
     }
   }
 
-  auto visitor = [comparator](const auto& aValue,
-                              const auto& bValue) -> ComparisonResult {
-    if constexpr (requires() { std::invoke(comparator, aValue, bValue); }) {
-      return fromBool(std::invoke(comparator, aValue, bValue));
-    } else {
-      AD_FAIL();
-    }
-  };
-
-  return ValueId::visitTwo(visitor, a, b);
+  return a.visit([&](const auto& aValue) {
+    return b.visit([&](const auto& bValue) -> ComparisonResult {
+      if constexpr (requires() { std::invoke(comparator, aValue, bValue); }) {
+        return fromBool(std::invoke(comparator, aValue, bValue));
+      } else {
+        AD_FAIL();
+      }
```

Comment:
With our updated compiler versions and consistent with a change that I have in the pipeline we can now simply write
```
return std::visit([&}(const auto& aValue, const auto& bValue) {...}, a, b);
```

Reply:
Okay, we can't because those are not `std::variant` at all, but I am currently merging the conflict there myself.